packages/nextjs/src/middleware/personalize-middleware.ts

```diff
@@ -12,34 +12,12 @@ import { CloudSDK } from '@sitecore-cloudsdk/core/server';
 import { personalize } from '@sitecore-cloudsdk/personalize/server';
 import { SitecoreConfig } from '../config';
 
-export type CdpServiceConfig = {
-  /**
-   * Your Sitecore Edge Platform endpoint
-   * Default is https://edge-platform.sitecorecloud.io
-   */
-  sitecoreEdgeUrl?: string;
-  /**
-   * Your unified Sitecore Edge Context Id
-   */
-  sitecoreEdgeContextId: string;
-  /**
-   * The Sitecore CDP channel to use for events. Uses 'WEB' by default.
-   */
-  channel?: string;
-  /**
-   * Currency for CDP request. Uses 'USA' as default.
-   */
-  currency?: string;
-  /**
-   * Timeout (ms) for CDP request. Default is 400.
-   */
-  timeout?: number;
-};
-
 export type PersonalizeMiddlewareConfig = MiddlewareBaseConfig &
   SitecoreConfig['api']['edge'] &
   SitecoreConfig['personalize'] & {
     clientFactory: GraphQLRequestClientFactory;
+    personalizeService?: GraphQLPersonalizeService;
+    getOverrideExperienceParams?: (req: NextRequest) => ExperienceParams;
```

Comment:
Not sure why we need to introduce this option as user can override getExperienceParams method.